packager/VisibilityChecker.cc

```diff
@@ -352,24 +352,46 @@ class VisibilityCheckerPass final {
             return tree;
         }
 
+        bool isExported = false;
+        if (lit.symbol.isClassOrModule()) {
+            isExported = lit.symbol.asClassOrModuleRef().data(ctx)->flags.isExported;
+        } else if (lit.symbol.isFieldOrStaticField()) {
+            isExported = lit.symbol.asFieldRef().data(ctx)->flags.isExported;
+        }
+
+        // Did we use a constant that wasn't exported?
+        if (!isExported) {
+            if (auto e = ctx.beginError(lit.loc, core::errors::Packager::UsedPackagePrivateName)) {
+                auto &pkg = ctx.state.packageDB().getPackageInfo(otherPackage);
+                e.setHeader("`{}` resolves but is not exported from `{}`", lit.symbol.show(ctx), pkg.show(ctx));
+                e.addErrorLine(lit.symbol.loc(ctx), "Defined here");
+                if (auto exp = pkg.addExport(ctx, lit.symbol)) {
+                    e.addAutocorrect(std::move(exp.value()));
+                }
+                if (!db.errorHint().empty()) {
+                    e.addErrorNote("{}", db.errorHint());
+                }
+            }
+
+            return tree;
+        }
+
```

Comment:
I moved this up, because it doesn't make sense to tell someone to import a package if after importing it, the thing won't be defined anyways.

If after exporting it it's still not imported, at least the "you need to import it" error message will now show the "exported from here" message.